### PR TITLE
New new BLE logging characteristic with LogRecord protos 

### PR DESCRIPTION
**Pull request by thebentern**

### CODE REVIEW

**Review by geeksville — APPROVED** [review]
Looks fine to me (except for the sizeof oops - I think you want to use length instead.

File: src/nimble/NimbleBluetooth.cpp [hunk 1]
```diff
@@ -268,12 +267,12 @@ void NimbleBluetooth::clearBonds()
     NimBLEDevice::deleteAllBonds();
 }
 
-void NimbleBluetooth::sendLog(const char *logMessage)
+void NimbleBluetooth::sendLog(const uint8_t *logMessage, size_t length)
 {
-    if (!bleServer || !isConnected() || strlen(logMessage) > 512) {
+    if (!bleServer || !isConnected() || sizeof(logMessage) > 512) {
```
[the remark this answers is not in the document]
Comment (thebentern) [reply src/nimble/NimbleBluetooth.cpp:272]: Doh! 😅

**Comment by geeksville** [comment]
After you merge to master I'll pull and spend a few hours doing the logging cleanup we discussed with @ianmcorvidae.  (I made a [commit](https://github.com/geeksville/Meshtastic-esp32/commit/51f8acbc5918709459aa980bf33387c45e9adaad) to start on this but I'll reapply after your changers are in.  I'd like to move all the Log stuff out of RedirectablePrint into its own singleton.  And have a chain of registered log handlers pass any generated logrecs out whenever we have clients connected (i.e. serial, BLE, TCP).

**Comment by thebentern** [comment]
> After you merge to master I'll pull and spend a few hours doing the logging cleanup we discussed with @ianmcorvidae. (I made a [commit](https://github.com/geeksville/Meshtastic-esp32/commit/51f8acbc5918709459aa980bf33387c45e9adaad) to start on this but I'll reapply after your changers are in. I'd like to move all the Log stuff out of RedirectablePrint into its own singleton. And have a chain of registered log handlers pass any generated logrecs out whenever we have clients connected (i.e. serial, BLE, TCP).
> 
> Cool with ya'll?

Sounds good to me. 
Are you done with the protobuf changes? I'm looking to land the plane and cut a protobuf version release if so.

**Review by garthvh — APPROVED** [review]

**Pull request merged**